packages/ast-spec/tests/util/typescript-estree-import.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Nx is picking up on the fact that we technically have a circular dependency between ast-spec
+ * and typescript-estree.
+ *
+ * This circular dependency only occurs in the tests/ for ast-spec and not in the main package source.
+ *
+ * We could therefore solve this by separating the ast-spec tests out into their own package, but the
+ * other option is to get Nx to turn a blind eye to the circular dependency by doing the following:
+ *
+ * - Removing @typescript-eslint/typescript-estree as an explicit devDependency in the package.json
+ * - Add an `.nxignore` file at the root of the monorepo which ignores this file (which we then in turn
+ * ensure is the only place we directly import from the @typescript-eslint/typescript-estree package).
+ */
```

Comment:
This feels like something we (@JamesHenry 😄) should feature request into Nx: the ability to have technical circular dependencies only in tests...

Reply:
I can't remember the exact history of this but I'm pretty sure I provided this breakdown and created this solution to the problem.

As it turns out on this PR right now the `@typescript-eslint/typescript-estree` package is not in the package.json of ast-spec, so we could simplify things to:

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/900523/164988392-53c7d8fa-def1-4b88-9a60-194ef93904a0.png">

The specifying of the package in the package.json is the thing that causes the rigid circular dependency to exist - there would be no way to know that the package was only used in certain situations when that is the way the relationship is defined. As part of the Nx updates I'm doing we'll be moving entirely to source to source relationships between packages/graph nodes (as is far more common) and we won't need to worry about this in that situation

Reply:
(I didn't include it in the screenshot but feel free to move some of the commentary around why we _don't_ want to include the package in the package.json to make that lint rule happy and that's why we are disabling it inline)

Reply:
Was there an update to nx to allow this? I swear there was a reason we went with using an `.nxignore` file.

Or did we used to rely on implicit dependency graph and now we have an explicit one configured it doesn't matter?